devutils/git-hooks/pre-commit

```diff
@@ -44,33 +51,61 @@ PreCommit::BuildChecks () {
         go install github.com/bazelbuild/buildifier/buildifier
     fi
 
-    output=$("$GOPATH/bin/buildifier" -mode=check $filepath)
+    output=$("$GOPATH/bin/buildifier" -mode=check "$filepath")
     if [ -n "$output" ]; then
         Log::Warning "$output"
-        Log::Failure "formatting error in a BUILD file. Aborting..."
+        Log::Error "formatting error in a BUILD file."
+    fi
+}
+
+
+# Rules checking file is correctly registered in the commit
+PreCommit::GitChecks () {
+    local filepath="$1"
+
+    # Ensure commit contains the same file that the one currently in the project
+    if [ "$(git status --porcelain "$filepath" | cut -c2)" != ' ' ]; then
+        Log::Error "unstaged change detected, but file is staged in the commit"
     fi
 }
 
 
 main () {
-    cd $(bazel info workspace)
+    cd "$(bazel info workspace)" || Log::Failure "unable to go in the workspace root. Aborting..."
 
     # Ensures every tests affected by modified targets pass
     git diff --name-only --staged HEAD \
         | Bazel::GetAssociatedTests \
         | Bazel::TestTargets || Log::Failure "test failure, aborting commit."
+    Log::Success "tests passed successfully!"
 
     # Per languages rules
-    git diff --name-only --staged HEAD | while read filepath
+    has_errors=false
+    while read -r filepath
     do
+        Log::ResetErrorCounter
         if [ -e "$filepath" ]; then
             case "$filepath" in
-                *.go) PreCommit::GoChecks $filepath || exit 1 ;;
-                *.ts) PreCommit::TypeScriptChecks $filepath || exit 1;;
-                *BUILD) PreCommit::BuildChecks $filepath || exit 1 ;;
+                *.go) PreCommit::GoChecks "$filepath";;
+                *.ts) PreCommit::TypeScriptChecks "$filepath";;
+                *BUILD) PreCommit::BuildChecks "$filepath";;
             esac
         fi
-    done
+        PreCommit::GitChecks "$filepath"
+
+        if [ "$(Log::GetErrorCounter)" -ne "0" ]; then
+            has_errors=true
+            Log::Error "file $filepath contains errors"
+        else
+            Log::Success "file $filepath is valid"
+        fi
+    done <<< "$(git diff --name-only --staged HEAD)"
+
+    if $has_errors; then
```

Comment:
I won't include the Log::Failure into the above statement, since it can exit the program while running tests on all files. The point is to show results of the linters / checkers on all file, and fail if one of them has an error.

However I made a one-liner for this.